File: backend/src/domain/user/StaffRole.ts
import {Clinic} from "@/domain/clinic";

export class StaffRole {
  constructor(
    public id?: string,
    public name?: string,
    public clinic?: Clinic,
    public rules?: string[],
  ) {
  }

  static listRules = new Set<string>([

  ])
}

File: backend/src/main/java/com/hypocrates/hypocrates/core/domain/staff/StaffRole.java
package com.hypocrates.hypocrates.core.domain.staff;

import com.hypocrates.hypocrates.core.domain.clinic.Clinic;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;


@Getter
@Builder
@AllArgsConstructor
public class StaffRole {
    private UUID id;
    private String name;

    private AppRule[] rules;

    static public StaffRoleBuilder OwnerBuilder() {
        return StaffRole.builder().name("Owner").rules(AppRule.values());
    }
}
